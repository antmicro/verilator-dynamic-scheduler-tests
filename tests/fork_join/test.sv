module t (/*AUTOARG*/
   // Inputs
   clk
   );
   input clk;

   initial begin
      fork
            #8 $write("[%0t] forked process 1\n", $time);
            #4 $write("[%0t] forked process 2\n", $time);
            #2 $write("[%0t] forked process 3\n", $time);
            $write("[%0t] forked process 4\n", $time);
      join
      #8 $write("[%0t] main process\n", $time);
      $write("*-* All Finished *-*\n");
      $finish;
   end
endmodule
