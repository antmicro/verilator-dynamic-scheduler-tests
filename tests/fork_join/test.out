forked process
forked process
forked process
main process
*-* All Finished *-*
